bazel/boringssl_fips.patch

```diff
@@ -0,0 +1,83 @@
+diff --git a/boringssl/ssl/handshake_client.cc b/boringssl/ssl/handshake_client.cc
+index 59ef6ec5f..5e9081367 100644
+--- a/boringssl/ssl/handshake_client.cc
++++ b/boringssl/ssl/handshake_client.cc
+@@ -232,18 +232,10 @@ static bool ssl_write_client_cipher_list(SSL_HANDSHAKE *hs, CBB *out) {
+   // Add TLS 1.3 ciphers. Order ChaCha20-Poly1305 relative to AES-GCM based on
+   // hardware support.
+   if (hs->max_version >= TLS1_3_VERSION) {
+-    if (!EVP_has_aes_hardware() &&
+-        !CBB_add_u16(&child, TLS1_CK_CHACHA20_POLY1305_SHA256 & 0xffff)) {
+-      return false;
+-    }
+     if (!CBB_add_u16(&child, TLS1_CK_AES_128_GCM_SHA256 & 0xffff) ||
+         !CBB_add_u16(&child, TLS1_CK_AES_256_GCM_SHA384 & 0xffff)) {
+       return false;
+     }
+-    if (EVP_has_aes_hardware() &&
+-        !CBB_add_u16(&child, TLS1_CK_CHACHA20_POLY1305_SHA256 & 0xffff)) {
+-      return false;
+-    }
+   }
+ 
+   if (hs->min_version < TLS1_3_VERSION) {
+diff --git a/boringssl/ssl/internal.h b/boringssl/ssl/internal.h
+index 16e100b6d..e44d8b602 100644
+--- a/boringssl/ssl/internal.h
++++ b/boringssl/ssl/internal.h
+@@ -670,6 +670,9 @@ const SSL_CIPHER *ssl_choose_tls13_cipher(CBS cipher_suites, uint16_t version,
+                                           uint16_t group_id);
+ 
+ 
++// ssl_tls13_cipher_meets_policy returns true if |cipher_id| is acceptable.
++bool ssl_tls13_cipher_meets_policy(uint16_t cipher_id);
++
+ // Transcript layer.
+ 
+ // SSLTranscript maintains the handshake transcript as a combination of a
+diff --git a/boringssl/ssl/s3_both.cc b/boringssl/ssl/s3_both.cc
+index 7ad821041..1a816f48a 100644
+--- a/boringssl/ssl/s3_both.cc
++++ b/boringssl/ssl/s3_both.cc
+@@ -691,6 +691,16 @@ class CipherScorer {
+   const bool security_128_is_fine_;
+ };
+ 
++
++bool ssl_tls13_cipher_meets_policy(uint16_t cipher_id) {
++  switch (cipher_id) {
```

Comment:
nit: How about just: `return cipher_id != TLS1_CK_CHACHA20_POLY1305_SHA256 & 0xffff`

Reply:
(Or is this currently implementation helpful for future integration?)

Reply:
I reverse engineered `set_compliance` method from the later boringssl - this method exists in new boringssl and is somewhat similar.